rtl/eclrtl/rtlformat.cpp

```diff
@@ -832,18 +832,12 @@ void CPropertyTreeWriter::outputQuoted(const char *text)
 
 void CPropertyTreeWriter::outputString(unsigned len, const char *field, const char *fieldname)
 {
-    unsigned olen = len;
     if (flags & XWFtrim)
         len = rtlTrimStrLen(len, field);
     if ((flags & XWFopt) && (rtlTrimStrLen(len, field) == 0))
         return;
-    if (olen != len)
-    {
-        StringBuffer tmp(len, field);
-        target->setProp(fieldname, tmp.str());
-    }
-    else
-        target->setProp(fieldname, field);
+    StringBuffer tmp(len, field);
+    target->setProp(fieldname, tmp.str());
```

Comment:
It is a bit of a shame to temporarily make a copy purely to ensure null terminated, for it then to be copied again inside the IPT call.
I did wonder whether I shouldn't introduce a IPropertyTree::setProp(const char *xpath, size32_t len, const char *val) .. and thereby avoid the need...

Reply:
Seems like a reasonable thing to add to me...

Reply:
(I suspect there are other places where it would have been used if available)

Reply:
A new function would make sense, but this change looks ok.